### DISCUSSION TITLE
Commits not always signed on self-hosted GitLab

### DISCUSSION THREAD

**Discussion by jokay** [renovatebot/renovate#9599] [Request Help] [answered]
If using self-hosted Renovate on self-hosted GitLab, the commit is not always signed.

**What Renovate type, platform and version are you using?**

- Self-hosted GitLab **13.10.3**
- Self-hosted Renovate **24.115.1** *(renovate/renovate:slim)*
- renovate.json

```json
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
        "config:base",
        "docker:enableMajor"
    ],
    "labels": [
        "dependencies"
    ]
}
```

**Describe the bug**

Commit for digest updates is not signed.

*(screenshot removed)*

Commit for digest update working.

*(screenshot removed)*

Commit for tag update not working.

*(screenshot removed)*

**Relevant debug logs**

Not sure if this may be of help as it can be reproduced?

*(but can provide logs if needed, once there are new digest updates)*

**Have you created a minimal reproduction repository?**

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

-

**Comment by rarkins** [comment]
Have you configured [gitPrivateKey](https://docs.renovatebot.com/self-hosted-configuration/#gitprivatekey)?

**Reply by jokay** [reply]
Yes, as environment variable:

```yml
RENOVATE_GIT_PRIVATE_KEY: -----BEGIN PGP PRIVATE KEY BLOCK-----***-----END PGP PRIVATE KEY BLOCK-----\n
```

**Comment by rarkins** [comment]
Showing screenshots of commits in your main branch is not helpful here either because renovate does not commit directly to the main branch, it commits to its own branches. You could be losing the signed commit during the merge, for example. You should be checking only the branches Renovate creates and commits to.

**Reply by jokay** [reply]
Ok, will remove them.

**Reply by rarkins** [reply]
If you find a commit within one of Renovate's branches that does not show up as verified then you need the debug logs from the run when that commit happened.

**Reply by rarkins** [reply]
It could be useful to also compare the logs to a branch/commit where it did work, to see what the difference is.

**Reply by jokay** [reply]
> Showing screenshots of commits in your main branch is not helpful here either because renovate does not commit directly to the main branch, it commits to its own branches. You could be losing the signed commit during the merge, for example. You should be checking only the branches Renovate creates and commits to.

But this commit is on the renovate branch.

![same](https://user-images.githubusercontent.com/18613935/115136757-2b41d900-a022-11eb-81e2-9004325366bc.png)

**Reply by rarkins** [reply]
Do you have the logs from this commit?

**Reply by jokay** [reply]
Sure, but it's not a debug log 😉 

```sh
$ docker-entrypoint.sh
 INFO: Autodiscovered repositories
       "length": 4,
       "repositories": ["user/repo3", "user/repo4", "user/repo2", "user/repo1"]
 INFO: No docker daemon found
 INFO: Repository started (repository=user/repo3)
       "renovateVersion": "24.115.1"
 INFO: Dependency extraction complete (repository=user/repo3)
       "baseBranch": "main",
       "stats": {
         "managers": {"gitlabci": {"fileCount": 1, "depCount": 3}},
         "total": {"fileCount": 1, "depCount": 3}
       }
 INFO: Repository finished (repository=user/repo3)
       "durationMs": 10645
 INFO: No docker daemon found
 INFO: Repository started (repository=user/repo4)
       "renovateVersion": "24.115.1"
 INFO: Dependency extraction complete (repository=user/repo4)
       "baseBranch": "main",
       "stats": {
         "managers": {
           "docker-compose": {"fileCount": 2, "depCount": 2},
           "dockerfile": {"fileCount": 1, "depCount": 3},
           "gitlabci": {"fileCount": 1, "depCount": 1},
           "nuget": {"fileCount": 2, "depCount": 31}
         },
         "total": {"fileCount": 6, "depCount": 37}
       }
 INFO: Repository finished (repository=user/repo4)
       "durationMs": 23033
 INFO: No docker daemon found
 INFO: Repository started (repository=user/repo2)
       "renovateVersion": "24.115.1"
 INFO: Dependency extraction complete (repository=user/repo2)
       "baseBranch": "main",
       "stats": {
         "managers": {"gitlabci": {"fileCount": 1, "depCount": 2}},
         "total": {"fileCount": 1, "depCount": 2}
       }
 INFO: Repository finished (repository=user/repo2)
       "durationMs": 3305
 INFO: No docker daemon found
 INFO: Repository started (repository=user/repo1)
       "renovateVersion": "24.115.1"
 INFO: Dependency extraction complete (repository=user/repo1)
       "baseBranch": "main",
       "stats": {
         "managers": {
           "docker-compose": {"fileCount": 28, "depCount": 52},
           "gitlabci": {"fileCount": 1, "depCount": 2}
         },
         "total": {"fileCount": 29, "depCount": 54}
       }
 INFO: Branch created (repository=user/repo1, branch=renovate/docker-docker.io-pihole-pihole-v5.8)
       "commitSha": "5233c15"
 INFO: PR created (repository=user/repo1, branch=renovate/docker-docker.io-pihole-pihole-v5.8)
       "pr": 355,
       "prTitle": "Update docker.io/pihole/pihole:v5.8 Docker digest to 6cf42db"
 INFO: Branch created (repository=user/repo1, branch=renovate/docker-docker.io-aler9-rtsp-simple-server-0.x)
       "commitSha": "b2a9906"
 INFO: PR created (repository=user/repo1, branch=renovate/docker-docker.io-aler9-rtsp-simple-server-0.x)
       "pr": 356,
       "prTitle": "Update docker.io/aler9/rtsp-simple-server Docker tag to v0.15.5"
 INFO: Repository finished (repository=user/repo1)
       "durationMs": 33963
Uploading artifacts for successful job
```

**Reply by viceice** [reply]
Why are you calling `docker-entrypoint.sh` ?

Please use our provided `renovate` command.

Checkout our prepared gitlab pipeline templates.

https://gitlab.com/renovate-bot/renovate-runner

**Reply by jokay** [reply]
Thx for the advice, I will have a look.

**Comment by jokay** [comment, marked as answer]
It may be due to the used **rebase** functionality in GitLab?

**Reply by viceice** [reply]
If you use the gitlab rebase it will create a new commit and old signature will be missing.

I don't know if there is a feature in gitlab to sign commits from ui.

**Reply by jokay** [reply]
Ok, then this is the answer for my problem 😉
